src/index.js

```diff
@@ -109,6 +109,14 @@ export class BrowserHistory extends History {
   }
 
   /**
+   * Returns the fully-qualified root of the current history object.
+   */
+  getAbsoluteRoot(): string {
+    let origin = createOrigin(this.location.protocol, this.location.hostname, this.location.port);
```

Comment:
Sorry I didn't think of this until I saw the router PR, but should this include a hash when not using pushstate?